### PR TITLE
Promises chaining

### PR DESCRIPTION
**Pull request by mahdiHash**
Please review and comment/commit if anything is wrong.

### CODE REVIEW

File: 1-js/11-async/03-promise-chaining/article.md [hunk 1]
```diff
@@ -181,19 +181,19 @@ loadScript("/article/promise-chaining/one.js").then(script1 => {
...
+کسانی که استفاده از promiseها را شروع می‌کنند گاهی اوقات درباره زنجیره‌سازی نمی‌دانند پس کد را اینگونه می‌نویسند. به طور کلی، زنجیره‌سازی ترجیح داده می‌شود.
 
-Sometimes it's ok to write `.then` directly, because the nested function has access to the outer scope. In the example above the most nested callback has access to all variables `script1`, `script2`, `script3`. But that's an exception rather than a rule.
+گاهی نوشتن `.then` به صورت مستقیم مشکلی ندارد چون تابع تودرتو به محدوده بیرونی دسترسی دارد. در مثال بالا تودرتوترین callback به تمام متغیر های `script1`، `script2` و `script3` دسترسی دارد. اما این بیشتر از آن که یک قانون باشد، یک استثنا است.
 
 
-````smart header="Thenables"
-To be precise, a handler may return not exactly a promise, but a so-called "thenable" object - an arbitrary object that has a method `.then`. It will be treated the same way as a promise.
+````smart header="Thenableها"
+اگر بخواهیم دقیق باشیم، یک مدیریت‌کننده ممکن است دقیقا یک promise برنگرداند اما شیءای به اصطلاح "thenable" را برگرداند - یک شیء دلخواه که متد `.then` را دارد. با این شیء درست مانند یک promise رفتار می‌شود.
```
Comment (mahdiHash) [inline 1-js/11-async/03-promise-chaining/article.md:192]: This line needs more attention.

File: 1-js/11-async/03-promise-chaining/article.md [hunk 2]
```diff
@@ -181,19 +181,19 @@ loadScript("/article/promise-chaining/one.js").then(script1 => {
...
 
-The idea is that 3rd-party libraries may implement "promise-compatible" objects of their own. They can have an extended set of methods, but also be compatible with native promises, because they implement `.then`.
+ایده این است که کتابخانه‌های شخص ثالث ممکن است شیءهای «سازگار با promise» خودشان را پیاده‌سازی کنند. این شیءها ممکن است مجموعه‌ای از متدهای خودشان را داشته باشند اما با promiseها نیز سازگار باشند چون آن‌ها `.then` را پیاده‌سازی می‌کنند.
```
Comment (mahdiHash) [inline 1-js/11-async/03-promise-chaining/article.md:194]: This line needs more attention.

File: 1-js/11-async/03-promise-chaining/article.md [hunk 3]
```diff
@@ -315,15 +315,15 @@ fetch('/article/promise-chaining/user.json')
...
-  // triggers after 3 seconds
+  // بعد از 3 ثانیه فعال می‌شود
   .then(githubUser => alert(`Finished showing ${githubUser.name}`));
 ```
 
-That is, the `.then` handler in line `(*)` now returns `new Promise`, that becomes settled only after the call of `resolve(githubUser)` in `setTimeout` `(**)`. The next `.then` in the chain will wait for that.
+یعنی اینکه مدیریت‌کننده .then در خط (*) حالا یک new Promise برمی‌گرداند که فقط بعد از فراخوانی resolve(githubUser) در setTimeout خط (**) تسویه می‌شود. .then بعدی در زنجیره برای آن صبر خواهد کرد.
 
-As a good practice, an asynchronous action should always return a promise. That makes it possible to plan actions after it; even if we don't plan to extend the chain now, we may need it later.
+به عنوان یک عادت خوب، یک عمل ناهنگام باید همیشه یک promise برگرداند. این باعث می‌شود که بتوان بعد از آن عملیاتی را برنامه‌ریزی کرد؛ حتی اگر نخواهیم زنجیره را الان گسترش دهیم، ممکن است بعدا به آن نیاز داشته باشیم.
```
Comment (mahdiHash) [inline 1-js/11-async/03-promise-chaining/article.md:324]: This line needs more attention.

File: 1-js/11-async/03-promise-chaining/article.md [hunk 4]
```diff
@@ -213,70 +213,70 @@ new Promise(resolve => resolve(1))
...
 ```
 
-This makes a network request to the `url` and returns a promise. The promise resolves with a `response` object when the remote server responds with headers, but *before the full response is downloaded*.
+این یک درخواست شبکه‌ای به url می‌فرستد و یک promise را برمی‌گرداند. زمانی که سرور همراه با headerها پاسخ می‌دهد، promise همراه با یک شیء response تسویه می‌شود اما *قبل از اینکه تمام پاسخ دانلود شود*.
 
-To read the full response, we should call the method `response.text()`: it returns a promise that resolves when the full text is downloaded from the remote server, with that text as a result.
+برای خواندن پاسخ کامل، ما باید متد response.text() را فراخوانی کنیم: این متد یک promise برمی‌گرداند که بعد از دانلود شدن کامل متن از سرور ریموت، همراه با متن به عنوان نتیجه resolve می‌شود.
 
-The code below makes a request to `user.json` and loads its text from the server:
+کد پایین یک درخواست به user.json می‌فرستد و متن آن را از سرور بارگیری می‌کند:
```
Comment (mahdyar) [inline 1-js/11-async/03-promise-chaining/article.md:239]: You removed backticks all over the file, it's just an example, please add them back:
```suggestion
کد پایین یک درخواست به `user.json` می‌فرستد و متن آن را از سرور بارگیری می‌کند:
```

File: 1-js/11-async/03-promise-chaining/article.md [hunk 5]
```diff
@@ -213,70 +213,70 @@ new Promise(resolve => resolve(1))
...
 
 ```js
 let promise = fetch(url);
 ```
 
-This makes a network request to the `url` and returns a promise. The promise resolves with a `response` object when the remote server responds with headers, but *before the full response is downloaded*.
+این یک درخواست شبکه‌ای به url می‌فرستد و یک promise را برمی‌گرداند. زمانی که سرور همراه با headerها پاسخ می‌دهد، promise همراه با یک شیء response تسویه می‌شود اما *قبل از اینکه تمام پاسخ دانلود شود*.
 
-To read the full response, we should call the method `response.text()`: it returns a promise that resolves when the full text is downloaded from the remote server, with that text as a result.
+برای خواندن پاسخ کامل، ما باید متد response.text() را فراخوانی کنیم: این متد یک promise برمی‌گرداند که بعد از دانلود شدن کامل متن از سرور ریموت، همراه با متن به عنوان نتیجه resolve می‌شود.
```
Comment (mahdyar) [inline 1-js/11-async/03-promise-chaining/article.md:237]: here's another one.

**Review by mahdyar — CHANGES_REQUESTED** [review]

**Comment by mahdiHash** [comment]
Thanks for your review, @mahdyar.
/done

**Review by mahdyar — APPROVED** [review]
Thank you, @mahdiHash!

**Pull request merged**